fvm/src/kernel/default.rs

```diff
@@ -1079,23 +1078,19 @@ where
                         .into(),
                 );
             }
+
             if header.key_len > MAX_KEY_LEN as u32 {
                 let tmp = header.key_len;
-                return Err(syscall_error!(IllegalArgument; "event key exceeded max size: {} > {MAX_KEY_LEN}", tmp).into());
+                return Err(syscall_error!(LimitExceeded; "event key exceeded max size: {} > {MAX_KEY_LEN}", tmp).into());
```

Comment:
nit, but in https://github.com/filecoin-project/FIPs/pull/831 it lists that we check each entry key individually for valid UTF8  while in our impl we do it all upfront for all keys.

Reply:
Gah, my patch reverts my previous fixes.

Reply:
Fixed.